Srcs/heiftojpeg/heiftojpeg.cpp

```diff
@@ -0,0 +1,183 @@
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::HVCC) {
+            cout << "HVCC ";
+            // cout << reader.getPropertyHvcc(contextId, property.index);
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::IMIR) {
+            cout << "IMIR ";
+            cout << reader.getPropertyImir(contextId, property.index).horizontalAxis;
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::IROT) {
+            cout << "IROT ";
+            rotation = reader.getPropertyIrot(contextId, property.index).rotation;
+            cout << rotation;
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::ISPE) {
+            cout << "ISPE ";
+            // cout << reader.getPropertyIspe(contextId, property.index);
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::LHVC) {
+            cout << "LHVC ";
+            // cout << reader.getPropertyLhvc(contextId, property.index);
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::LSEL) {
+            cout << "LSEL ";
+            cout << reader.getPropertyLsel(contextId, property.index).layerId;
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::OINF) {
+            cout << "OINF ";
+            // cout << reader.getPropertyOinf(contextId, property.index);
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::RLOC) {
+            cout << "RLOC ";
+            cout << reader.getPropertyRloc(contextId, property.index).horizontalOffset;
+        } else if (property.type == ImageFileReaderInterface::ItemPropertyType::TOLS) {
+            cout << "TOLS ";
+            cout << reader.getPropertyTols(contextId, property.index).targetOlsIndex;
+        } else {
+            cout << "unknown ";
+        }
+        cout << "\n";
+    }
+
+    // Always reuse the parameter set from the first tile, sometimes tile 7 or 8 is corrupted
+    HevcImageFileReader::ParameterSetMap parameterSet;
+    reader.getDecoderParameterSets(contextId, tileItemIds.at(0), parameterSet);
+    std::string codeType = reader.getDecoderCodeType(contextId, tileItemIds.at(0));
+    ImageFileReaderInterface::DataVector parametersData;
+    if ((codeType == "hvc1") || (codeType == "lhv1")) {
+        // VPS (HEVC specific)
+        parametersData.insert(parametersData.end(), parameterSet.at("VPS").begin(), parameterSet.at("VPS").end());
+    }
+
+    if ((codeType == "avc1") || (codeType == "hvc1") || (codeType == "lhv1")) {
+        // SPS and PPS
+        parametersData.insert(parametersData.end(), parameterSet.at("SPS").begin(), parameterSet.at("SPS").end());
+        parametersData.insert(parametersData.end(), parameterSet.at("PPS").begin(), parameterSet.at("PPS").end());
+    } else {
+        // No other code types supported
+        // throw FileReaderException(FileReaderException::StatusCode::UNSUPPORTED_CODE_TYPE);
```

Comment:
Should we throw here? If not the block can be removed.

Reply:
yes - this code was copied out of HevcFileReader and I wasn't yet sure how to namespace everything. this has been finished now